docs/dev/plugin-quickstart.rst

```diff
@@ -0,0 +1,914 @@
+        logger: logging.Logger = logger,
+    ):
+        """
+        The mymodel model class MyModelModel.
+
+        Contains methods to read, write, setup and update mymodel models.
+
+        Parameters
+        ----------
+        root : str, Path, optional
+            Path to the model folder
+        mode : {'w', 'w+', 'r', 'r+'}, optional
+            Mode to open the model, by default 'w'
+        config_fn : str, Path, optional
+            Path to the model configuration file, by default None to read
+            from template in build mode or default name in update mode.
+        data_libs : list of str, optional
+            List of data catalogs to use, by default None.
+        logger : logging.Logger, optional
+            Logger to use, by default logger
+        """
+        # Add model _CATALOGS to the data_libs, if you wish to use default catalogs
+        if self._CATALOGS:
+            if isinstance(data_libs, str):
+                data_libs = [data_libs]
+            if data_libs is None:
+                data_libs = []
+            data_libs = data_libs + self._CATALOGS
+
+        # Call the parent class initialisation method using super()
+        super().__init__(
+            root=root,
+            mode=mode,
+            config_fn=config_fn,
+            data_libs=data_libs,
+            logger=logger,
+        )
+
+        # If your model needs any extra specific initialisation parameter or step
+        # add them here
+        # Example: an extra propery or default value, or a new HydroMT component
+
+
+        # end of the __init__ function
+
+Model components
+^^^^^^^^^^^^^^^^
+A reminder from the :ref:`Model guide <model_interface>`:
+
+  HydroMT defines any model through the model-agnostic Model API based on several *general* components and *computational* unit components.
+  Each component represents a specific model data type and is parsed to a specific Python data object.
+  The general components are **maps** (raster data), **geoms** (vector data), **forcing**, **results**, **states**, and **config**
+  (the model simulation configuration). These are available to all model classes and plugins.
+  The computational components are different for different types of models: i.e. **grid** for distributed or grid models,
+  **response_units** for lumped or semi-distributed models, **mesh** for mesh or unstructured grid models, and **network** for
+  network models (to be developed).
+
+So typically, when you want to add data to your model (DEM, precipitation, parameters etc.), you add data to one of the
+specific Model component. To add data, to a component, the ``set_<component>`` methods are you use and each of the components can be
+read and written using the specific ``read_<component>`` and ``write_<component>`` methods.
+
+The table below lists the base model components common to all model classes.
+All base model attributes and methods can be found in the :ref:`API reference <model_api>`
```

Comment:
Super useful, but why not linking to the same table in the model_overview.rst
I'm a bit afraid of maintenance and clarity of the docs if the same thing is stated in (many) different places.